indigo_api/timeline.py

```diff
@@ -219,6 +226,13 @@ def get_timeline(work, only_approved_events=False):
         # even though the timeline is given in reverse chronological order,
         # each date on the timeline is described in regular order: assent first, repeal last
 
+        # chapter number (start of validity)
+        for chapter_number in all_chapter_numbers.filter(validity_start_date=date):
```

Comment:
this goes to the DB each time through the outer for loop. Instead, load all the chapter numbers once and filter them in python.